allauth/socialaccount/providers/apple/provider.py

```diff
@@ -0,0 +1,27 @@
+from allauth.account.models import EmailAddress
+from allauth.socialaccount.providers.base import ProviderAccount
+from allauth.socialaccount.providers.oauth2.provider import OAuth2Provider
+
+
+class AppleProvider(OAuth2Provider):
+    id = 'apple'
+    name = 'Apple'
+    account_class = ProviderAccount
+
+    def extract_uid(self, data):
+        return str(data['sub'])
+
+    def extract_common_fields(self, data):
+        return dict(
+            email=data.get('email'),
+        )
+
+    def extract_email_addresses(self, data):
+        ret = []
+        email = data.get('email')
+        if email and data.get('email_verified'):
+            ret.append(EmailAddress(email=email, verified=True, primary=True))
```

Comment:
If there is an email, we should always add it to the list, even if not verified. 
```suggestion
            ret.append(EmailAddress(email=email, verified=email.get('email_verified'), primary=True))
```

Reply:
@pennersr updated by this [commit](https://github.com/pennersr/django-allauth/pull/2424/commits/25d094299f426d892a04fe521c7d186fe5510745)